tracee-rules/signatures/rego/regosig/traceerego.go

```diff
@@ -35,7 +35,7 @@ const queryMetadata string = "data.%s.__rego_metadoc__"
 const packageNameRegex string = `package\s.*`
 
 // NewRegoSignature creates a new RegoSignature with the provided rego code string
-func NewRegoSignature(regoCodes ...string) (types.Signature, error) {
+func NewRegoSignature(target string, partialEval bool, regoCodes ...string) (types.Signature, error) {
```

Comment:
NIT Not sure how many args is too much, but at some point we may consider [functional options](https://dave.cheney.net/2014/10/17/functional-options-for-friendly-apis) pattern for friendly API

Reply:
IMO think 3 (considering regoCodes is a variadic) is still fine, although we could take the functional approach but could do as a separate PR.

Reply:
Yup. We can refactor to use functional Opts later on.